src/main/java/org/tron/keystore/WalletFile.java

```diff
@@ -0,0 +1,461 @@
+        // Also add the following to the ObjectMapperFactory
+        // objectMapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+        public void setKdfparams(KdfParams kdfparams) {
+            this.kdfparams = kdfparams;
+        }
+
+        public String getMac() {
+            return mac;
+        }
+
+        public void setMac(String mac) {
+            this.mac = mac;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o.getClass() != Crypto.class) {
+                return false;
+            }
+            
+            Crypto that = (Crypto) o;
+            
+            if (getCipher() != null
+                    ? !getCipher().equals(that.getCipher())
+                    : that.getCipher() != null) {
+                return false;
+            }
+            if (getCiphertext() != null
+                    ? !getCiphertext().equals(that.getCiphertext())
+                    : that.getCiphertext() != null) {
+                return false;
+            }
+            if (getCipherparams() != null
+                    ? !getCipherparams().equals(that.getCipherparams())
+                    : that.getCipherparams() != null) {
+                return false;
+            }
+            if (getKdf() != null
+                    ? !getKdf().equals(that.getKdf())
+                    : that.getKdf() != null) {
+                return false;
+            }
+            if (getKdfparams() != null
+                    ? !getKdfparams().equals(that.getKdfparams())
+                    : that.getKdfparams() != null) {
+                return false;
+            }
+            return getMac() != null
+                    ? getMac().equals(that.getMac()) : that.getMac() == null;
+        }  
+
+        @Override
+        public int hashCode() {
+            int result = getCipher() != null ? getCipher().hashCode() : 0;
+            result = 31 * result + (getCiphertext() != null ? getCiphertext().hashCode() : 0);
+            result = 31 * result + (getCipherparams() != null ? getCipherparams().hashCode() : 0);
+            result = 31 * result + (getKdf() != null ? getKdf().hashCode() : 0);
+            result = 31 * result + (getKdfparams() != null ? getKdfparams().hashCode() : 0);
+            result = 31 * result + (getMac() != null ? getMac().hashCode() : 0);
+            return result;
+        }
+        
+    }
+
+    public static class CipherParams {
+        private String iv;
+
+        public CipherParams() {
+        }
+
+        public String getIv() {
+            return iv;
+        }
+
+        public void setIv(String iv) {
+            this.iv = iv;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o.getClass() != CipherParams.class) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Compare to "this.getClass()" instead. ([squid:S2162](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2162))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23iWkDisRwWBfuxI&open=AWPK23iWkDisRwWBfuxI)

[ISSUE_KEY]: AWPK23iWkDisRwWBfuxI